### PR TITLE
Deltavision: new date format and native units

### PR DESCRIPTION
**Pull request by sbesson**
See https://trello.com/c/y5FN422r/61-deltavision-reader-date-parsing

With this PR included,:
- Deltavision datasets like the one liked from the card should have their date correctly parsed by Bio-Formats 
- working distance should be stored using the native unit (MM)

To test this PR:
- check the automated tests pass agains the repository with the new sample DV file included
- open the sample file with `showinf -nopix -omexml-only ` and check that no warning is output regarding the date format, the `AcquisitionDate` is populated in the OME-XML metadata and the `Objective` `WorkingDistanceUnit` is expressed in `mm`.

### CODE REVIEW

**Comment by melissalinkert** [comment]
All looks good to me, just waiting on builds that include the last commit.

File: components/formats-gpl/src/loci/formats/in/DeltavisionReader.java [hunk 1]
```diff
@@ -64,7 +64,12 @@
   public static final int DV_MAGIC_BYTES_1 = 0xa0c0;
   public static final int DV_MAGIC_BYTES_2 = 0xc0a0;
 
+  /* Deprecated in favor of DATE_FORMATS */
   public static final String DATE_FORMAT = "E MMM d HH:mm:ss yyyy";
+  public static final String[] DATE_FORMATS = {
```
Comment (joshmoore) [inline components/formats-gpl/src/loci/formats/in/DeltavisionReader.java]: This is a breaking change. Perhaps a second "ALTERNATIVE_DATE_FORMATS" array?

Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/DeltavisionReader.java]: :+1: for `@Depreacat`ing - will push a commit

Comment (sbesson) [inline components/formats-gpl/src/loci/formats/in/DeltavisionReader.java]: See https://github.com/sbesson/bioformats/commit/a1af388996b3953b3327704281ab697809bf8f3b

**Comment by melissalinkert** [comment]
I didn't notice any new deprecation warnings, and https://ci.openmicroscopy.org/view/Bio-Formats/job/BIOFORMATS-DEV-merge-full-repository/ is now green.

**Comment by bramalingam** [comment]
--rebased-to #2292

**Pull request merged**